src/AppBundle/Command/AddUserCommand.php

```diff
@@ -0,0 +1,259 @@
+        if (null === $email = $input->getArgument('email')) {
+            $question = new Question(' > <info>Email</>: ');
+            $question->setValidator(array($this, 'emailValidator'));
+            $question->setMaxAttempts(self::MAX_ATTEMPTS);
+
+            $email = $console->ask($input, $output, $question);
+            $input->setArgument('email', $email);
+        } else {
+            $output->writeln(' > <info>Email</>: '.$email);
+        }
+    }
+
+    /**
+     * This method is executed after the interact() and before the execute()
+     * method. It's main purpose is to initialize the variables used in the rest
+     * of the command methods.
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->em = $this->getContainer()->get('doctrine')->getManager();
+    }
+
+    /**
+     * This method is executed after interact() and initialize(). It usually
+     * contains the logic to execute to complete this command task.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $startTime = microtime(true);
+
+        // first check if a user with the same username already exists
+        $existingUser = $this->em->getRepository('AppBundle:User')->findOneBy(array(
+            'username' => $username = $input->getArgument('username'),
+        ));
+
+        if (null !== $existingUser) {
+            throw new \RuntimeException(sprintf('There is already a user registered with the "%s" username.', $username));
+        }
+
+        $this->username = $input->getArgument('username');
+        $this->plainPassword = $input->getArgument('password');
+        $this->email = $input->getArgument('email');
+        $this->isAdmin = $input->getOption('is-admin');
+
+        // create the user and encode its password
+        $user = new User();
+        $user->setUsername($this->username);
+        $user->setEmail($this->email);
+        $user->setRoles(array($this->isAdmin ? 'ROLE_ADMIN' : 'ROLE_USER'));
+
+        // See http://symfony.com/doc/current/book/security.html#security-encoding-password
+        $encoder = $this->getContainer()->get('security.password_encoder');
+        $encodedPassword = $encoder->encodePassword($user, $this->plainPassword);
+        $user->setPassword($encodedPassword);
+
+        $this->em->persist($user);
+        $this->em->flush($user);
+
+        $output->writeln('');
+        $output->writeln(sprintf('[OK] %s was successfully created: %s (%s)', $this->isAdmin ? 'Administrator user' : 'User', $user->getUsername(), $user->getEmail()));
+
+        if ($output->isVerbose()) {
+            $finishTime = microtime(true);
+            $elapsedTime = $finishTime - $startTime;
+
+            $output->writeln(sprintf('[INFO] New user database id: %d / Elapsed time: %.2f ms', $user->getId(), $elapsedTime*1000));
+        }
+    }
+
+    public function passwordValidator($plainPassword)
```

Comment:
I would tag them as ``@internal`` with a comment saying they are public for PHP 5.3 compat